geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/registry/ProtobufOperationContextRegistry.java

```diff
@@ -138,5 +142,10 @@ private void addContexts() {
             // Resource permissions get handled per-function, since they have varying permission
             // requirements.
             new ResourcePermission(ResourcePermission.NULL, ResourcePermission.NULL)));
+    operationContexts.put(MessageTypeCase.OQLQUERYREQUEST,
+        new ProtobufOperationContext<>(ClientProtocol.Message::getOqlQueryRequest,
+            new OqlQueryRequestOperationHandler(),
+            opsResp -> ClientProtocol.Message.newBuilder().setOqlQueryResponse(opsResp),
+            new ResourcePermission(Resource.DATA, Operation.READ)));
```

Comment:
Even though OQL statements can result in the execution of Java methods, is it standard to use `READ` permissions for queries?

Reply:
Actually, queries can't invoke arbitrary methods - see GEODE-3247. So READ is the correct permission - that's what the old protocol and gfsh and are checking.